pkr/driver/buildx.py

```diff
@@ -0,0 +1,282 @@
+          * no_rebuild: do not build if destination image exists
+        """
+        if docker is None:
+            # Handle python 3.6 here, to not impact child drivers
+            raise Exception("buildx is not supported for python < 3.6")
+
+        services = services or list(self.kard.env.get_container().keys())
+        if rebuild_context:
+            self.kard.make()
+
+        buildx_meta = self.kard.meta.get("buildx", {})
+        if "cache_registry_username" in buildx_meta and buildx_meta["cache_registry"] is not None:
+            registry_url = buildx_meta["cache_registry"].split("/")[0]
+            print(f"Logging to {registry_url}")
+            docker.login(
+                server=registry_url,
+                username=buildx_meta.get("cache_registry_username", None),
+                password=buildx_meta.get("cache_registry_password", None),
+            )
+        self._create_builder(purge=clean_builder)
+
+        tag = tag or self.kard.meta["tag"]
+
+        if parallel:
+            if len(services) > 1:
+                write("Building docker images using {} threads ...\n".format(parallel))
+            futures = []
+            with ProcessPoolExecutor(max_workers=parallel) as executor:
+                for service in services:
+                    futures.append(
+                        executor.submit(
+                            *self._build_image(
+                                service, tag, verbose, logfile, nocache, no_rebuild, True
+                            )
+                        )
+                    )
+            for future in futures:
+                future.result(timeout=1800)
+        else:
+            if len(services) > 1:
+                write("Building docker images...\n")
+            for service in services:
+                exec = self._build_image(
+                    service, tag, verbose, logfile, nocache, no_rebuild, False
+                )
+                if exec is not None:
+                    exec[0](*exec[1:])
+
+    def _build_image(
+        self,
+        service,
+        tag=None,
+        verbose=True,
+        logfile=None,
+        nocache=False,
+        no_rebuild=False,
+        bufferize=None,
+    ):
+        """Build docker image.
+
+        Args:
+          * service: service to build
+          * tag: the tag on which the image will be saved
+          * verbose: verbose logs
+          * logfile: separate log file for the underlying build
+          * nocache: disable docker cache
+          * no_rebuild: do not build if destination image exists
+          * bufferize: keep log to print when ended
+        """
+        image_name = self.make_image_name(service, tag)
+
+        dockerfile = self.kard.env.get_container(service).get("dockerfile")
+        if not dockerfile:
+            return
+
+        if no_rebuild:
+            image = len(self.docker.images(image_name)) == 1
+
+        if not no_rebuild or image is False:
```

Comment:
image might not be defined if no_rebuild is False

Reply:
That's intended.